### PR TITLE
simple-cipher: add test template

### LINKED ISSUE #1992: simple-cipher: add test template
## Task
Create a test template file for `simple-cipher`

### Details
1. If this is your first time contributing to this repository, welcome!
    - Contributing to a repository on GitHub requires a [fork](https://help.github.com/en/articles/fork-a-repo)
1. Create a branch on your local repository called `simple-cipher-add-test-template`
    - Run: `git checkout -b simple-cipher-add-test-template`
1. Clone problem-specifications repository; this is used to populate the template
    - Run: `git clone https://github.com/exercism/problem-specifications ../problem-specifications`
1. Create `exercises/simple-cipher/.meta/template.j2`
    - See [Generator Documentation] for more information
1. Generate test file from new template
    - Run: `bin/generate_tests.py simple-cipher`

At this point, if the generator fails, or you are otherwise stuck,
please feel free to reach out! It is perfectly acceptable to submit
an incomplete PR in order to get feedback or assistance.

### Resources
- [Contributing Guide]
- [Generator Documentation]
- [Closing issues using keywords]

[Contributing Guide]: https://github.com/exercism/python/blob/master/docs/CONTRIBUTING.md
[Generator Documentation]: https://github.com/exercism/python/blob/master/docs/GENERATOR.md
[Closing issues using keywords]: https://help.github.com/en/articles/closing-issues-using-keywords

### PR DESCRIPTION
**Pull request by beatorizu**
Add template to generate simple-cipher tests. To generate the python code I update some settings of `canonical-data.json`, because it was using other methods to compare string length.

Resolves #1992

### CODE REVIEW

File: exercises/simple-cipher/.meta/template.j2 [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{%- import "generator_macros.j2" as macros with context -%}
+import re
+{{ macros.header(imports=['Cipher']) }}
+
```
Comment (cmccandless) [inline exercises/simple-cipher/.meta/template.j2]: To make this work, you'll need to apply the following changes to `bin/generate_tests.py`. This is a filter that should've been there from the beginning, and I have no problem adding it as part of this PR:
```
diff --git a/bin/generate_tests.py b/bin/generate_tests.py
index 3a602cf..61475c8 100755
--- a/bin/generate_tests.py
+++ b/bin/generate_tests.py
@@ -95,6 +95,10 @@ def has_error_case(cases):
     return False
 
 
+def regex_replace(s, find, repl):
+    return re.sub(find, repl, s)
+
+
 def load_canonical(exercise, spec_path):
     """
     Loads the canonical data for an exercise as a nested dictionary
@@ -206,6 +210,7 @@ def generate(
     env = Environment(loader=loader, keep_trailing_newline=True)
     env.filters['to_snake'] = to_snake
     env.filters['camel_case'] = camel_case
+    env.filters['regex_replace'] = regex_replace
     env.tests['error_case'] = error_case
     result = True
     for exercise in glob(os.path.join('exercises', exercise_glob)):
```

Suggested Change:
```suggestion
{%- macro convert_js(s) -%}
{{ s |
    regex_replace("\\.substring\\((\\d+), ([^)]+)\\)", "[\\1:\\2]") |
    regex_replace("(\w+)\\.length", "len(\\1)")
}}
{%- endmacro %}

```

File: exercises/simple-cipher/.meta/template.j2 [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            {% if "cipher.key" in expected -%}
+            {% set expected = expected -%}
+            {% elif expected == plaintext -%}
+            {% set expected = 'plaintext' -%}
+            {% elif expected == key -%}
+            {% set expected = 'cipher.key' -%}
+            {% else -%}
+            {% set expected = "'" ~ expected ~ "'" -%}
+            {% endif -%}
+            {% if plaintext -%}
```
Comment (cmccandless) [inline exercises/simple-cipher/.meta/template.j2]: Use the suggested new macro:
```suggestion
            {%- set expected = convert_js(expected) -%}
            {% if plaintext -%}
```

File: exercises/simple-cipher/.meta/template.j2 [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            {% elif property == "decode" -%}
+            {% if "cipher.key" in ciphertext -%}
+            {% set ciphertext = ciphertext -%}
+            {% elif "cipher.encode" in ciphertext -%}
+            {% set ciphertext = ciphertext ~ '(plaintext)' -%}
+            {% elif ciphertext == key -%}
+            {% set ciphertext = 'cipher.key' -%}
+            {% else -%}
+            {% set ciphertext = "'" ~ ciphertext ~ "'" -%}
+            {% endif -%}
```
Comment (cmccandless) [inline exercises/simple-cipher/.meta/template.j2]: Use above suggested macro:
```suggestion
            {% endif -%}
            {%- set ciphertext = convert_js(ciphertext).replace("expected", expected) -%}
```

**Comment by yawpitch** [comment]
Renaming this so we can easily see what exercise it relates to.

**Comment by beatorizu** [comment]
Suggestions applied :D

File: exercises/simple-cipher/.meta/template.j2 [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+{{ macros.header(imports=['Cipher']) }}
+{%- macro convert_js(s) -%}
+{{ s |
+    regex_replace("\\.substring\\((\\d+), ([^)]+)\\)", "[\\1:\\2]") |
+    regex_replace("(\w+)\\.length", "len(\\1)")
+}}
+{%- endmacro %}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+{{ macros.utility() }}
```
Comment (cmccandless) [inline exercises/simple-cipher/.meta/template.j2]: Not used.

```suggestion
```

**Pull request merged**